AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -177,16 +173,16 @@ - (void)generatePlaceholderImage
 - (void)imageAtTime:(CMTime)imageTime completionHandler:(void(^)(UIImage *image))completionHandler
 {
   ASPerformBlockOnBackgroundThread(^{
-    ASDN::MutexLocker l(_videoLock);
+    AVAsset *asset = self.asset;
```

Comment:
`tracksWithMediaCharacteristic:` blocks the current thread (background thread in this case) until it loads the required metadata from the asset. That loading may be over a network. If `_videoLock` is lock on a background thread that's waiting on the network, other threads, such as the UI thread are blocked attempting to acquire the lock.  In order to prevent the blocking while the metadata is loaded, the lock needs to be limited in scope. 

Repeated access of the `asset` variable is the reason for the original lock. This change makes a local snapshot of the `asset` variable to ensure it's consistent throughout the method. A data consistency check made in the completion block of imageAtTime:

``` obj-c
// Ensure the asset hasn't changed since the image request was made
if (ASAssetIsEqual(weakSelf.asset, asset)) {
  [weakSelf setVideoPlaceholderImage:image];
}
```

If the `asset` is changed while the placeholder image is being generated, the image is ignored. Preventing the incorrect image from displaying.

I can change `imageAtTime` to be a static variable with that accepts an asset as a parameter. That way it doesn't access any instance variables and it's clear it doesn't need to acquire _videoLock.